lib/pure/json.nim

```diff
@@ -531,6 +531,21 @@ proc `[]`*(node: JsonNode, index: BackwardsIndex): JsonNode {.inline, since: (1,
 
   `[]`(node, node.len - int(index))
 
+proc `[]`*[U, V](a: JsonNode, x: HSlice[U, V]): JsonNode =
+  ## Slice operation for JArray.
+  ##
+  ## Returns the inclusive range `[a[x.a], a[x.b]]`:
+  runnableExamples:
+    let arr = %[0,1,2,3,4,5]
+    doAssert arr[2..4] == %[2,3,4]
+    doAssert arr[2..^2] == %[2,3,4]
+
+  assert(a.kind == JArray)
+  result = newJArray()
+  let xb = (when x.b is BackwardsIndex: a.len - int(x.b) else: int(x.b))
+  for i in x.a .. xb:
```

Comment:
what if `x.a` is a BackwardsIndex? You need to copy the logic from system.nim precisely. ;-)